doc/adr/release.md

```diff
@@ -0,0 +1,1019 @@
+
+In the release configuration file, a `flake-output-path` is a value that
+represents a node from a path in a derivation accessible from a project's flake
+`outputs`. To use the node, the derivation needs to be built first.
+
+A `flake-output-path` value can take one of the forms...
+
+1. A string representing a selector of a derivation from the flake's `outputs`
+2. A map containing the keys
+    - `output` - A string representing a selector of a derivation from the flake's
+    `outputs`.
+    - `path` - An optional file path that will be used to select a file from the
+    derivation once it has been built.
+
+Examples
+
+```yaml
+- apps.x86_64-linux.foo
+- output: devShell.x86_64-linux
+- output: packages.x86_64-linux.bar
+  path: /bin/bar
+- output: packages.x86_64-linux.bizz
+  path: /lib
+```
+
+#### Intrinsics
+
+##### Indicators
+
+###### Allways publish git tag (Indicator)
+
+When [`git.tag.always-publish`](#git.tag.always-publish) is `True`, include
+[Tag exists on GitHub](#Tag-exist-on-GitHub).
+
+###### Allways publish GitHub release (Indicator)
+
+When [`github.release.always-publish`](#github.release.always-publish) is `True`,
+include [Release exists on GitHub](#release-exists-on-github).
+
+###### Assets to publish (Indicator)
+
+When [`github.release.assets`](#github.release.assets) is not the Empty Map,
+include [Release exists on GitHub](#release-exists-on-github).
+
+##### Objectives
+
+###### Release exists on GitHub
+
+A release is on GitHub for the HEAD commit, tagged with the release tag, with
+the appropriate contents.
+
+Check if release exists on GitHub
+
+- If it does not exist, achieve via [Create GitHub release](#create-github-release)
+- If it does exist, not achievable.
+
+###### Tag exists on GitHub
+
+A git tag for the HEAD commit, with the appropriate contents, is publish to
+public git repositories.
+
+- Check if tag exists on GitHub
+  - If it does not exist, achieve via [Push remote tag](#push-remote-tag).
+  - If it does exist, check if it is as expected
+    - If it is, achieved.
+    - If it is not, not achievable.
+
+###### GitHub Cli configured correctly
+
+- Check if the GitHub Cli is configured correctly
+  - If it is, achieved.
+  - If it is not, not achievable.
+
+###### Tag exists locally
+
+- Check if tag exists locally
+  - If it does not exists, achieve via [Create local tag](#create-local-tag).
+  - If it does exists, check if it is as expected
+    - If it is, achieved.
+    - If it is not, not achievable.
+
+###### Git configured correctly
+
+- Check if git is configured correctly
+  - If it is, achieved.
+  - If it is not, not achievable.
+
+##### Changes
+
+###### Create GitHub release
+
+Create a release on appropriate GitHub repo.
+
+- Preconditions
+  - [Git configured correctly](#git-configured-correctly)
+  - [GitHub Cli configured correctly](#github-cli-configured-correctly)
+  - [Tag exists on GitHub](#tag-exists-on-github)
+- Actions
+
+  ```console
```

Comment:
indentation

Reply:
I was not sure what a good way was to make a multi-link code-block a list item.

Reply:
Removed.